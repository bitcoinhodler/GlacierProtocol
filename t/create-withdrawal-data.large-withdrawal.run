#!/bin/bash
set -e

# Test large withdrawal transactions. This spends 7 inputs from an older
# Glacier address with uncompressed, non-segwit keys.


# Simulate an earlier, even-larger transaction. These files should
# get deleted by glacierscript.
touch transaction-01.png transaction-02.png transaction-03.png


$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2MyCc3oEVwSYzgmspKTjZ6HZrzeg8ajdmKR
52410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54ae
bcrt1qrwxllgvtc0ns624dlped894kkfq608jmye2k50
7
0200000000010117cc27e3f5042a6502e5faec7fc12e44dc933fce75ca7971fb6fb481df3561600100000000ffffffff0242e419000000000017a914385ceedba0a97aa2d9ae88f8140ddd9524e6cdd88740420f000000000017a9144151ec0c95327edca1c4c5d38d01e00d9ec93d57870247304402200714f6af838f4d1df221095fb0e22a185de733c8f16b1ef3b65773c1c05e085b022047225e704c3ceb8b8d83d5a6b5824154baf3fb691e2c3ebe5152d1d11a3dc78c012102f5455327f645f6b6c44fbaf03d56fd65cba828961239960a3284cd43a889057300000000
02000000000101be30567e2748b8315d44ede1e729e7144f2c53b3f83e62561a4d8e67477d0abc0100000017160014713a9050f95ee7ce6da75ecd2713bc809f184118ffffffff02866072000000000017a914f3fadb6de38d8a2f3cfac4c96eb6822ed88e44b38740420f000000000017a9144151ec0c95327edca1c4c5d38d01e00d9ec93d578702473044022022ae9f7c93c32f4746d82140a7958ba617567b34c07bfe7452cb5ccfa8dda3c102204fa070ccb0cdd76d42dc2c6c31e32675bdd35337aed9bb82e6ab5942940e2200012102a19f9119892ec588beacf5c39fed2c37bdd7318037e88e394052ccc86858129700000000
020000000001010e00de4cedad146e5762aae8f6f6c395e597a4373f0db26b47c0581ade429e700100000000ffffffff0290c4c0010000000017a91406eef6c60b644bb80f28af6953bc407fb70757fa8750690f000000000017a9144151ec0c95327edca1c4c5d38d01e00d9ec93d5787024730440220565f2ac28f5972035dcd16cf5a35bd4d41424af9bf502cbdc36d99840b6a6daf02204d3c2f2b660024adf7b9092a0b06130210727520f15d0b51a94b63f0e204d8fb0121029860dc0d45559ab5a2ef437f9ff1ec4f867c5b9fe30cb3dee408aa2d11ad4c2700000000
0200000000010109ad1ea5e3ec29d432e528c517abce3819715e3b83cd0ea5d399d4082f4344dd0100000017160014023e4b8e3e3b3432beecf3ebe95876690dbfbd1fffffffff0280de0f000000000017a9144151ec0c95327edca1c4c5d38d01e00d9ec93d5787608162000000000017a914b831ebd24f0a47a9e610e173f9e93083b6f87868870247304402201e7d2b423b6388297f05cb114b09b92cd72e8fd5c53d63c3a8a1cb1872095bd002200305c60eb6c61013f57d714521be6d6d8dc112bfda53cca5547ae40271b41736012102830fb7998dd6d3a4686fe31832e9255ea5efafd79448a5b5f0958ad5cdbbcd5a00000000
02000000000101e1e0d3683e53e5b286900db418b64854f93159b15b6068931e7b5938184c3cf301000000171600148a6c1aa3af05f9da1d05545d7f2634c468f90df8ffffffff022a7b52000000000017a914b330855b088486013ed13bf2cfd134c516f48cda87900510000000000017a9144151ec0c95327edca1c4c5d38d01e00d9ec93d57870247304402206d6abd9c63ccbce641a34fdd52cea0f58e0da9735dd525d66e16825401b7989a02205de70c5c3f990365d4a22f2ce5622e120bd8c9da7143cae1ea2e9cc9912f1fed012103595c5f5649e7a92aab69a9b1e9c1e6b8a486de71de804488499c2c6fbe41d30200000000
020000000001015856f2736edf390a301ac21d430c733a789ce4535424d8d6d8e66c23c901851d0100000017160014e47ceb06bc19e2ca4327b1cd7fe9c30b7266a323ffffffff02a02c10000000000017a9144151ec0c95327edca1c4c5d38d01e00d9ec93d57874a97b0010000000017a9141a4274c41f51b370c0370b4dd58608575f36c146870247304402203b0eb00cc07729f4e3456807289fbe1150678f783b0dcd736c66855523818fee022049546a04dc41a3c1b1859d03124ae2509b4747eac207e95ef3732ba9f46f7e17012103db82b75ec69868dd2d9d54652b8c0c081c72249a36c62db4b653cfb51bd208d600000000
02000000000101430b7e02c7ae4fc4c4b95b041b23f469bf7b4b6e221cc59803432bb7bd053b5201000000171600149dd2579d7b67fa89a69030220ed0a03efc7c9043ffffffff02b05310000000000017a9144151ec0c95327edca1c4c5d38d01e00d9ec93d5787f442a0010000000017a914e62ccee61b66e8ee7062018a1db17294b22396c887024730440220078f40f35cc44c5ce0fc60575924c27e6e377b582cbb58c491ccbb749ff3d6e502204d584ae427c8513528fb51fbe51875ef078005e0e707c3baf180b9a695f9eaa5012102e0ff4bd2ea8884f978b80dfd0ce9e5b0aa211942471180874a002172e7694d0400000000
2
93VDtjySSuPjgLYyCR6BBTqCuaqu6X728ZfkNEDbgAdVURRd1rS
924WjH7yMYyuyhUougBHEu2vThcHRmahF6FsKvNVQBwSGuEyqKm
1
y

y
INPUT

ls -1 transaction*.png
