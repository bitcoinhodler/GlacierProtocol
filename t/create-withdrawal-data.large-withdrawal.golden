Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #2 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #3 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #4 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #5 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #6 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #7 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.07230000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00003279 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00003279, you have 0.07226721 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to bcrt1qrwxllgvtc0ns624dlped894kkfq608jmye2k50 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.07226753 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.07230000 BTC in unspent inputs from cold storage address 2MyCc3oEVwSYzgmspKTjZ6HZrzeg8ajdmKR
0.07226753 BTC going to destination address bcrt1qrwxllgvtc0ns624dlped894kkfq608jmye2k50
Fee amount: 0.00003247

Signing with private keys: 
93VDtjySSuPjgLYyCR6BBTqCuaqu6X728ZfkNEDbgAdVURRd1rS
924WjH7yMYyuyhUougBHEu2vThcHRmahF6FsKvNVQBwSGuEyqKm


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 1.00 satoshis per vbyte

Raw signed transaction (hex):
0200000007e2ba88f4a0d52ba8e54c660ebe7bb585c03f5f445fa5c25dfab969c8dce667f001000000fd9f01004730440220294f169e1692b9ab7c4608cd4bf1fd86bfd2e3efcf26931c7165843ca9a0bc5b022013dea7f384abf6bece4c28d7b05cb96d3091bdcdc8cfe575cc6942c216ed0fc601473044022028dab3a21f2dd09b8910e9b249064cef58e8305ff3230f6683947502d7aa4f2202201f1fa2d1803b9a345ac0e357281cffe8d641da8302c41ba9853e7554932c7eb0014d0b0152410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54aeffffffffc3c63121ecb0abc64f8bdbcb019956917c0ed1ef1d20f9edf84924aba6e2961c01000000fd9f010047304402206230083f7c461f583f23d74aafbb4dcb114242697d6bc198b97e054965c96d25022017542974bd1ff085344438f1bd3f7519c69fdf705659e654b32acb8d1139fd710147304402202e9cd33a627ed2f9927add820f93a353b059df70b99308f21334d5edae9262fe02203e443940f29a7cfbdbc4f2a7044e90e19a30c8d619002109c234ecd3c0648871014d0b0152410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54aeffffffffbf0e0391c69519fcf062a11f5819bd8fa5579d6a3973ae3b7fa20d6a17cd079401000000fd9f010047304402201a8a045ba3ca30606b5bda2023e0ed66d9b3c94a902549e6ef8b875602627be702205f615729c8cd6ee4a9b832f40aadd422499b1505dd1e766b16b8847af06181f50147304402203ad8316c6dd561aca259b09b1349a6e0e592a05c7456b303b065b2b8ea8ce11e0220133a83c161129ef29acc4dded192bc12cb822617aac37c5e240613adef8a967f014d0b0152410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54aeffffffffaaa27862e63e4a8f07acd90b134c5daae8b3917e329bb2b9944ab93a28578bfc00000000fd9f010047304402200c42fa213d695fca12a311feeb9b9526c2d34e68db6389013c1ec5736b873f0002207136fa4305d6eccc2b8e9fde5b940c593f0e487570d8c7f47a7f29abcfd967270147304402205490ea9feee3cd4ebab54456d282b60fb946d7d8c54ca1589860ae6b5ab3785502202e8114fb6daad46ff24335da98a8faff056b08d6e99e3162a70a4830cef58895014d0b0152410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54aeffffffff3c35b39a29343aa950f22a651afb2a667c2d8188d41e696bc3234545735800c501000000fd9f0100473044022021e0517c29d1292f3e3c5ad4946e7936501c72afcbdc2ca07900076a305878dc022047294ddb09d27d5ab5b48d934b35b1770cbd28ad177552277eb3424a8336c2c101473044022059e1ee526b03691b784351d936a403a6bd2a1e8b0d25cc075bbdc8d373120d0602204d13ee719cdfd93fcd57375e6f57603d0148fc99ffc0f5a44a6486690f7b5f02014d0b0152410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54aeffffffff813553f40d42a0c10c601f05eabc3d245afbfbea1ead2d9451098caf9826e20000000000fd9f0100473044022045d76a456bf28945e8d7ade7738a443338c9ad744ce9d3e3ea744b337cb671f1022065f0446e3f0a2166927b9423dae5f4d23f9b201af5e411120b31104fab7d41510147304402202ed71eb9491a764cf6582716fef4d44c0e1ca55507378942ce76178015f1a8e002200629cdadbad1ebc4bb285e8ba2dc40286bb51c0694a38284259625af295ff1b9014d0b0152410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54aeffffffff68237326b3a24f396d757d8cbb5dbf1d4a800cb1ff761116c7f6050ec63c51bc00000000fd9f01004730440220474bfa8eea90a66bcf0ddcb5703ecc390cbf7c084eca379d59babefa1569aa7e02207e6537b2bbfc2b5ac01358327ff9e56ca07a7eb9c0e2586404bf74b763552e810147304402207766b5c75f00f3d25ec89066d839bc649c1519655d1644d70f65bb8a5d9b9f3902200a24ecf41fd5339b328c522cb6f4f89bdcff7c2fdef0aebabb5d35e22dd992da014d0b0152410498137b6223a9fa395e16fe6e49a7f4a737029d4476ec35844a8469e358d641b85a5bf355d570699d63cde87da1d450e1966d97ee61a99b51995a11eb7ea0f6744104fd9a69f95e31c3b8d2c97925e0f151dd1c67f3b71e3ba6db8f20d944e6e94d9b5e3ad4c12a5f9bd6d2137cf69b015f9637de61aafae29fdfccf8ef5a4cd81e374104ff188a131ee79900a4cdbe68c5ea84ad73b8acf8414e5895f6c7b0e4779ef50d39ba369a34cd3f7216877769a292ccf52c2653c799a138321bf8c1e03a9120594104eeafc0621008662a9977b953ee7f7af87981830f561cac50387211c75d58bcb751414fc9abe0cf4a28a4044806ccd54b3f80ba990e0eff046bb1b930968b973a54aeffffffff0181456e00000000001600141b8dffa18bc3e70d2aadf872d396b6b241a79e5b00000000

Transaction fingerprint (md5):
13a673b34ee229b8d449c936ffd489fb
QR code for transaction written to transaction-01.png,transaction-02.png
transaction-01.png
transaction-02.png
